internal/provisioner/kops_provisioner.go

```diff
@@ -8,146 +8,105 @@ import (
 	"github.com/pkg/errors"
 	log "github.com/sirupsen/logrus"
 
-	"github.com/mattermost/mattermost-cloud/internal/api"
 	"github.com/mattermost/mattermost-cloud/internal/model"
 	"github.com/mattermost/mattermost-cloud/internal/tools/kops"
 	"github.com/mattermost/mattermost-cloud/internal/tools/terraform"
 )
 
 // KopsProvisioner provisions clusters using kops+terraform.
 type KopsProvisioner struct {
-	clusterRootDir   string
-	s3StateStore     string
-	cs               clusterStore
-	kopsFactory      kopsFactoryFunc
-	terraformFactory terraformFactoryFunc
-	logger           log.FieldLogger
+	clusterRootDir string
+	s3StateStore   string
+	logger         log.FieldLogger
 }
 
 // NewKopsProvisioner creates a new KopsProvisioner.
-//
-// kopsFactory and terraformFactory exist purely to allow for unit testing. Passing nil defaults
-// to using the kops and terraform binary wrappers.
-func NewKopsProvisioner(clusterRootDir string, s3StateStore string, cs clusterStore, kopsFactory kopsFactoryFunc, terraformFactory terraformFactoryFunc, logger log.FieldLogger) *KopsProvisioner {
-	if kopsFactory == nil {
-		kopsFactory = func(logger log.FieldLogger) (KopsCmd, error) {
-			return kops.New(s3StateStore, logger)
-		}
-	}
-
-	if terraformFactory == nil {
-		terraformFactory = func(outputDir string, logger log.FieldLogger) TerraformCmd {
-			return terraform.New(outputDir, logger)
-		}
-	}
-
+func NewKopsProvisioner(clusterRootDir string, s3StateStore string, logger log.FieldLogger) *KopsProvisioner {
```

Comment:
I ripped out all the factory stuff that I had just added, since the new model doesn't require me to fake out the provisioner anymore, and we're not testing this code path right now anyway.

The provisioner now doesn't interact with the cluster store at all -- ideally, we can keep it focussed on just "do this" vs. "query database and decide to do this". I envision breaking this creation up into more states from which we can resume successfully after an interruption.

Reply:
Awesome!